Allow builing arbitrary rapid tags

WybrenKoelmans opened the pull request:
Fixes #2

WybrenKoelmans committed:
Allow builing arbitrary rapid tags
Fixes #2

WybrenKoelmans commented:
Code should be ready, just going to have to find out how to build / test this

p2004a commented:
assign me as reviewer so I get notified once it's ready

geekingfrog commented:
With this code change I then ran the following:

```bash
mkdir -p /tmp/rapid-store/
cd $BYAR-CHOBBY-REPO/
rapid-buildgit "$PWD" / modinfo.lua /tmp/rapid-store/chobby "$(git rev-parse HEAD)" chobby custom-tag custom-version
```
and then: `zcat /tmp/rapid-store/chobby/versions.gz` gives

```bash
chobby:custom-tag,2679b6fefa8d1be6d5a2a8e23c7c206e,,BYAR Chobby custom-version
chobby:test,22c581a391abb875e8e112ebb56daf98,,BYAR Chobby test-4449-e7131a6
```

Invoking the same `rapid-buildgit` command without tag nor version adds a new archive as usual:

```bash
chobby:custom-tag,2679b6fefa8d1be6d5a2a8e23c7c206e,,BYAR Chobby custom-version
chobby:git:e7131a69be76a48842958bbd6c04ecb174bca071,22c581a391abb875e8e112ebb56daf98,,BYAR Chobby test-4449-e7131a6
chobby:test,22c581a391abb875e8e112ebb56daf98,,BYAR Chobby test-4449-e7131a6
```

so that seems to work.

p2004a committed:
Revert incorrect script update

p2004a committed:
Refactor the new parameters to override tag and version

- Use std::optional for optional
- Adjust variable naming
- Better usage message
- Encapsulate the version resolution in correct functino
- The default version, when version override is set must depend
  on the branch, so we don't create test- versions outside of
  test branch
- Don't create package for versions from custom branches

p2004a commented:
@geekingfrog mostly

I've spend time to refactor it and fix some issues, most importantly without version override, the generated version should not have `test-` prefix, but the used custom rapid branch name.
